### ISSUE TITLE
AngularJS states as operations- telemetry not captured

### ISSUE THREAD

**Issue by koosala** [microsoft/ApplicationInsights-JS#950]
I had the [issue#923](https://github.com/microsoft/ApplicationInsights-JS/issues/923) which was closed because I was late in replying, but the solution did not work unfortunately. Since I cannot reopen the issue, adding a new one, sorry. The original issue will show my query. Here is the continuation.

appInsight.addTelemetryInitializer is underfined, and I could find the API only on the context property, similar to the [stackoverflow ](https://stackoverflow.com/questions/48508883/how-to-set-application-insights-operation-name-from-javascript) answer. Also, if you see the browserTelemetry.txt file I have attached in the original query, the telemetry seems to be carrying the operation name correctly, just not showing up in azure portal.

**Comment by markwolff** [comment]
According to the file you uploaded, you are using `a.0.js`, and so the way you've added the telemetry initializer is correct. Can you clarify what you mean by the operation name is not showing up in the portal correctly? Can you provide a screenshot, etc?

**Comment by koosala** [comment]
Thanks for confirming the first part.

I ran through some steps to capture the latest info - 
[browserTelemetry2.txt](https://github.com/microsoft/ApplicationInsights-JS/files/3342431/browserTelemetry2.txt)


It is a SPA application, I first navigate to the root element '/', at which time no routing has taken place. the first telemetry is sent.

The I navigate using the menu to an angularjs state, set the operation id and name, and there are Rest APIs called, on waiting for a few seconds, the second telemetry is sent, with the correct set operation id and name

Then I navigate to a second UI using state routing again, update the operation id and name. Some more api is called and the third telemetry is sent.

In Azure monitoring portal, the browser performance only shows the root element as an operation lasting 26 seconds, which feels like it is including the time of the two navigations. The other two operations don't show up. Azure monitoring screenshots from browser performance and transaction view attached: 
![BrowserPerformance](https://user-images.githubusercontent.com/5226969/60391051-64ceea80-9b03-11e9-8345-249b6ead2423.png)
![Transaction](https://user-images.githubusercontent.com/5226969/60391053-6a2c3500-9b03-11e9-847a-79de99e033a0.PNG)

**Comment by koosala** [comment]
Ok, I think I have resolved the issues mostly with a little more experience using the javascript API. Essentially, these were the changes:
1. My earlier approach had updated trackdependency calls with new operation id and name but there was no corresponding trackpage call with that that operation id. So, I had to first trigger a track page on view change

2. Additionally broke the track page call into a starttrack and stoptrack calls so that duration is captured correctly. That said, the view for the pageview is slightly skewed, as being discussed in [https://github.com/Microsoft/ApplicationInsights-JS/issues/781 ](url)

3. Updated appInsights.context.operation.id and appInsights.context.operation.name so that it is passed along with the dependency call through the REST API calls and into the server side.

4. Also set the enableCorsCorrelation true as UI and server APIs are hosted at different websites. Allowed for the server side correlation.

I guess there were pieces of documentation in different places, helped me get this going. Reviewing the telemetry closely really helped.
